src/message_subscriber.cpp

```diff
@@ -144,6 +147,28 @@ code message_subscriber::load(message_type type, uint32_t version,
         CASE_HANDLE_MESSAGE(stream, version, transaction);
         CASE_HANDLE_MESSAGE(stream, version, verack);
         CASE_HANDLE_MESSAGE(stream, version, version);
+        case message_type::block:
+        {
+            const auto message = std::make_shared<message::block>(settings);
+
+            // Subscribers are invoked only with stop and success codes.
+            if (!message->from_data(version, stream))
+                return error::bad_stream;
+
+            block_subscriber_->invoke(error::success, message);
+            return error::success;
+        }
+        case message_type::headers:
+        {
+            const auto message = std::make_shared<message::headers>();
+
+            // Subscribers are invoked only with stop and success codes.
+            if (!message->from_data(version, stream, settings))
+                return error::bad_stream;
+
+            headers_subscriber_->relay(error::success, message);
+            return error::success;
+        }
```

Comment:
I have inlined these two cases because they occur just once each. @evoskuil if you prefer I can also wrap them with a macro and template but I deemed it to be an overkill.

Reply:
Please handle in the same manner as CASE_RELAY_MESSAGE_SETTINGS. We should probably provide overloads for each message type so that these are not special cases and so that settings are already propagated for future cases.

Reply:
Done.